R/classify-diabetes.R

```diff
@@ -0,0 +1,105 @@
+#'   [tibble::tibble()] type object.
+#' @export
+#' @seealso See the [vignette("function-flow", package = "osdc")] for a detailed
+#'   description of the internal implementation of this classification function.
+#'
+#' @examples
+#' classify_diabetes(
+#'   kontakter = register_data$kontakter,
+#'   diagnoser = register_data$diagnoser,
+#'   lpr_diag = register_data$lpr_diag,
+#'   lpr_adm = register_data$lpr_adm,
+#'   sysi = register_data$sysi,
+#'   sssy = register_data$sssy,
+#'   lab_forsker = register_data$lab_forsker,
+#'   bef = register_data$bef,
+#'   lmdb = register_data$lmdb
+#' )
+classify_diabetes <- function(kontakter, diagnoser, lpr_diag, lpr_adm, sysi, sssy, lab_forsker, bef, lmdb) {
+  # Verification step -----
+  verify_required_variables(kontakter, "kontakter")
+  verify_required_variables(diagnoser, "diagnoser")
+  verify_required_variables(lpr_diag, "lpr_diag")
+  verify_required_variables(lpr_adm, "lpr_adm")
+  verify_required_variables(sysi, "sysi")
+  verify_required_variables(sssy, "sssy")
+  verify_required_variables(lab_forsker, "lab_forsker")
+  verify_required_variables(bef, "bef")
+  verify_required_variables(lmdb, "lmdb")
+
+  # Initially processing -----
+  lpr2 <- join_lpr2(
+    lpr_diag = lpr_diag,
+    lpr_adm = lpr_adm
+  )
+
+  lpr3 <- join_lpr3(
+    kontakter = kontakter,
+    diagnoser = diagnoser
+  )
+
+  # pregnancy_dates <-  get_pregrancy_dates(
+  #   lpr2 = lpr2,
+  #   lpr3 = lpr3
+  # )
+
+  # Inclusion steps -----
+  # diabetes_diagnosis <-  include_diabetes_diagnosis(
+  #   lpr2 = lpr2,
+  #   lpr3 = lpr3
+  # )
+
+  # podiatrist_services <- include_podiatrist_services(
+  #   sysi = sysi,
+  #   sssy = sssy
+  # )
+
+  # gld_purchases <- include_gld_purchases(
+  #   lmdb = lmdb
+  # )
+
+  included_hba1c <- include_hba1c(
+    lab_forsker = lab_forsker
+  )
+
+  # Exclusion steps -----
+  exclusions <- gld_purchases |>
+    exclude_potential_pcos(bef = bef) |>
+    exclude_wld_purchases(lmdb = lmdb) |>
+    exclude_pregnancy(
+      # TODO: Need to think about arg naming here..
+      hba1c = included_hba1c,
+      pregnancy_dates = pregnancy_dates
+    )
+
+  # Joining into an initial dataset -----
+  # inclusions <- join_inclusions(
+  #   included_diabetes_diagnosis,
+  #   included_podiatrist_services,
+  #   exclusions
+  # )
+
+  # inclusions |>
+  #   get_diagnosis_dates() |>
+  #   classify_t1d()
+}
+
+#' After inclusion and exclusion, classify those with type 1 diabetes.
+#'
+#' @param data Joined data output from the inclusion and exclusion steps.
+#'
+#' @return The same object type as the input data, which would be a
+#'   [tibble::tibble()] type object.
+#' @keywords internal
+#'
+classify_t1d <- function(data) {
+  # data |>
+  #   get_has_t1d_primary_diagnosis() |>
+  #   get_only_insulin_purchases() |>
+  #   get_majority_of_t1d_primary_diagnosis() |>
+  #   get_insulin_purchases_within_180_days() |>
+  #   get_insulin_is_two_thirds_of_gld_purchases()
+}
```

Comment:
If it is still present in the `classify_diabetes()` function environment, then adding the `gld_hba1c_after_exclusions` object as an input to `get_insulin_purchases_within_180_days()` would allow us to reuse the post-exclusion GLD data to assess insulin purchases in relation to the date of diabetes variable. This could be a way to solve what we discussed earlier today.

Reply:
The function outputs and helper functions currently described in #133 need a bit of aligning to match what ends up being implemented in this script (right now I've added another helper function, get_type_diagnosis_majority(), but that one only relates to hospital diagnoses, not GLD, so it should be fine for this PR). There will probably be some back-and-forth between the function flow docs and this script as we go along.